### ISSUE TITLE
IComparable implementation toggle

### ISSUE THREAD

**Issue by dmitriyi-affinity** [SteveDunn/Vogen#407]
### Describe the feature

I want to wrap a string value object.
But I don't want the struct to be comparable.
```CSharp
[ValueObject<string>]
public partial struct AccessToken
{
}
// Generated signature:
public partial struct AccessToken: global:: System.IEquatable<AccessToken>, global::System.IEquatable<System.String> ,  global::System.IComparable<AccessToken>, global::System.IComparable;

// Generated signature, if the new toggle "switch off IComparable" is on:
public partial struct AccessToken: global:: System.IEquatable<AccessToken>, global::System.IEquatable<System.String>;

```
It should not be too hard to implement.

**Comment by SteveDunn** [comment]
Thanks for the report @dmitriyi-affinity! It seems like a reasonable suggestion, but may I ask why you don't want them to be `IComparable`?

**Comment by dmitriyi-affinity** [comment]
The best examples are opaque tokens. Let's say access token. You have a reason to make them typed rather than strings. And if the MyAccessToken is inheriting IComparable it confuses people who first time seeing the class and your codebase.

**Comment by SteveDunn** [comment]
For the first stab at this, I added a parameter to `VogenConfigurationAttribute` (for global (assembly) config) and `ValueObjectAttribute` (for instance config). There parameter is called `comparison` and is of type `ComparisonGeneration`, which is either `Omit` or `UseUnderlying`. `UseUnderlying` is the default as having it as `Omit` would be a breaking change.

Currently seeing what else needs changing or where tests are needed. The branch is https://github.com/SteveDunn/Vogen/pull/463 if you're interested.

**Comment by SteveDunn** [comment]
Fixed and released in version 3.0.21
